macros/edr/materializations/tests/test.sql

```diff
@@ -0,0 +1,238 @@
+{% macro query_test_result_rows(sample_limit=none) %}
+  {%- set query -%}
+      select * from ({{ sql }}) {%- if sample_limit -%} limit {{ sample_limit }} {%- endif -%}
+  {%- endset -%}
+  {% do return(elementary.agate_to_dicts(dbt.run_query(query))) %}
+{% endmacro %}
+
+{% macro cache_elementary_test_results_rows(elementary_test_results_rows) %}
+  {% do elementary.get_cache("elementary_test_results").update({model.unique_id: elementary_test_results_rows}) %}
+{% endmacro %}
+
+
+{% macro handle_anomaly_test(flattened_test) %}
+  {% set metrics_tables_cache = elementary.get_cache("tables").get("metrics") %}
+  {% set metrics_table = elementary.get_elementary_test_table(flattened_test.name, 'metrics') %}
+  {% if metrics_table %}
+    {% do metrics_tables_cache.append(metrics_table) %}
+  {% endif %}
+
+  {% set anomaly_scores_rows = elementary.query_test_result_rows() %}
+  {% if anomaly_scores_rows and flattened_test.short_name == "dimension_anomalies" %}
+    {% set sample_row = anomaly_scores_rows[0] %}
+    {% do elementary.cache_elementary_test_results_rows([elementary.get_anomaly_test_result_row(flattened_test, sample_row, anomaly_scores_rows)]) %}
+  {% else %}
+    {% set test_result_metrics_rows = {} %}
+    {% for anomaly_score_row in anomaly_scores_rows %}
+      {% do test_result_metrics_rows.setdefault(anomaly_score_row.metric_name, []) %}
+      {% do test_result_metrics_rows[anomaly_score_row.metric_name].append(anomaly_score_row) %}
+    {% endfor %}
+
+    {% set elementary_test_results_rows = [] %}
```

Comment:
This is not compatible with the [current logic](https://github.com/elementary-data/dbt-data-reliability/blob/3ec6eef10fb4004dbea68561b8395b394f983fe2/macros/edr/tests/on_run_end/handle_tests_results.sql#L28) in which we create a row for each failure or for the latest anomaly scores.